cea/demand/thermal_loads.py

```diff
@@ -321,13 +323,18 @@ def calc_Qhs_sys(bpr, tsd):
     return tsd
 
 
-def calc_Qhs_Qcs(bpr, date, tsd, use_dynamic_infiltration_calculation, region):
+def calc_Qhs_Qcs(bpr, date, tsd, use_dynamic_infiltration_calculation, region, building_name, config, locator):
```

Comment:
calculation of set points seems to be independent of any other calculations - I think it should be refactored _out_ of `calc_Qhs_Qcs`. Also, I believe the check for the  predefined temperatures file should be outside of the routine that checks for it (a separate function) so that you don't need to call the simple version from inside that. this reduces the complexity of the function signatures as well as making the flow easier.

Reply:
Hey @daren-thomas , I do not understand your comment. Set point temperatures are further used for calculating the thermal loads. So I think the calculations should be part of the `calc_Qhs_Qcs`.

Reply:
hey @BhargavaKrishnaSreepathi what I mean is to pass the set point temperatures as a _parameter_ instead of adding all the parameters to calculate it instead. You need two functions, one computes whether you can actually do this varying setpoint calculation (true if files are present and the proper config variable is set) and the other function to perform the calculation.

That removes the reponsibility of calculating these setpoints from the `calc_Qhs_Qcs` function. Basically, a function should do one thing only.

Reply:
hmmm...but the original `calc_Qhs_Qcs` already had the temperature calculations within the function, I maintained the same structure. Unless you are talking of the removing the entirety of the temperature calculations, i.e. `calc_set_point_from_predefined_file` and `control_heating_cooling_systems.calc_simple_temp_control`, out of the ` calc_Qhs_Qcs`.

Reply:
@BhargavaKrishnaSreepathi yep. exactly. and make the set points a parameter to `calc_Qhs_Qcs`. basically, tidying up as we go. knowing how these set points were calculated is not `calc_Qhs_Qcs`'s responsibility.